fix:  integration docs (#79)
ethereum-optimism/community-hub: src/docs/developers/integration.md

> This process involves two primary sub-steps:

Reviewer: ```suggestion
This process involves three steps:
```

Reviewer: Fewer extraneous words! :)

1. Compile your contracts with the OVM Solidity compiler.
2. Deploy your contracts against a local instance of Optimistic Etheereum and our Kovan testnet.
3. Test your contracts against the local instance and Kovan testnet.

If you're using [hardhat](https://hardhat.org), our preferred development environment, this can be done with the use of a handy plugin we've built.
For the rest of this documentation, we'll expect you to have your Solidity contracts ready to go.
Please note that Vyper support is currently not enabled but is planned for a future release.

::: tip The Official Tutorial that you can fork!

> If you want to try out an example contract instead of deploying your own, you can follow our [optimism-tutorial](https://github.com/ethereum-optimism/optimism-tutorial) which serves as a quick and easy example to start bootstrap your project with.

Reviewer: ```suggestion
If you want to try out an example contract instead of deploying your own, you can follow our [optimism-tutorial](https://github.com/ethereum-optimism/optimism-tutorial) which serves as a quick and easy example to bootstrap your project with.
```

:::

> If you're using another testing suite like [Truffle](https://www.trufflesuite.com/) or [Waffle](https://getwaffle.io), that tutorial won't apply. But these `npm` packages have got you covered:

Reviewer: ```suggestion
If you're using another framework like [Truffle](https://www.trufflesuite.com/) or [Waffle](https://getwaffle.io), that tutorial won't apply. But these `npm` packages have got you covered:
```

- [`@eth-optimism/solc`](https://www.npmjs.com/package/@eth-optimism/solc): exports the Optimistic Ethereum compiler for `solidity@0.5/0.6/0.7`
- [`@eth-optimism/hardhat-ovm`](https://www.npmjs.com/package/@eth-optimism/hardhat-ovm): exports `artifacts-ovm` folder of contract artifacts which will work with contracts output by the compiler.

> An example of usage of Waffle with can be found in our [`Waffle-ERC20-Example`](https://github.com/ethereum-optimism/Waffle-ERC20-Example) repository which serves as a simple guide for how to get started using Waffle for Optimistic Ethereum.

Reviewer: ```suggestion
An example of using Waffle on Optimistic Ethereum can be found in our [`Waffle-ERC20-Example`](https://github.com/ethereum-optimism/Waffle-ERC20-Example) repository.
```

1. [High level overview](https://hackmd.io/elr0znYORiOMSTtfPJVAaA) of differences.
2. [Complete EVM/OVM comparison](/docs/protocol/evm-comparison) of all discrepancies.

### Adding and installing dependencies

> In this step, we will assume that you have a pre-existing Node.js project that you would to turn in to an Optimistic Ethereum project.

Reviewer: ```suggestion
In this step, we will assume that you have a pre-existing Ethereum project built with Node.js, that you would like to convert to an Optimistic Ethereum project.
```

      accounts: {
        mnemonic: 'test test test test test test test test test test test junk'
      },
      gasPrice: 0,
      ovm: true
    }
  },
  solidity: '0.7.6', // Must match solc version of the OVM compiler
  ovm: {
    solcVersion: '0.7.6' // Currently, we only support 0.5.16, 0.6.12, and 0.7.6 of the Solidity compiler
  },
  namedAccounts: {
    deployer: 0
  },
}
```

The network changes that we added here let us use Hardhat's `network` flag to control which network we wish to compile, test, and deploy our contracts for.
Pretty neat!

Now we're ready to move on to compiling your contracts.

## Step 1. Compiling contracts

We're going to use our `@eth-optimism/hardhat-ovm` package (described in the previous section above).
If this package is not already in your project, let's install this package now:

```sh
yarn add @eth-optimism/hardhat-ovm
```

This package now let's us compile our contracts for the OVM.
But first, let compile your contracts for L1.
Run the following command to do so:

```sh
yarn hardhat compile
```

To ensure that your contracts have been compiled correctly, check your project directory for the outputted folders `artifacts` and `cache`.

> To compile your contacts for L2, we merely add the `network` flag to our command and specify the network we wish to compile for:

Reviewer: ```suggestion
To compile your contacts for L2, we simply add the `network` flag to our command and specify the network we wish to compile for:
```
